### PR TITLE
Add mypy testrunner

### PR DESCRIPTION
**Pull request by marvin-w**
Yay!

We should add new modules to tox.ini one by one and fix the upcoming errors by hand or with monkeytype :-D.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -31,3 +31,21 @@ known_third_party = xknx
...
+python_version = 3.7
+ignore_errors = true
+follow_imports = silent
+ignore_missing_imports = true
+warn_incomplete_stub = true
+warn_redundant_casts = true
+warn_unused_configs = true
+
+# add the modules below once we add typing for them so that we fail the build in the future if someone changes something without updating the typings
+[mypy-xknx.telegram.*]
```
Comment (marvin-w) [inline setup.cfg:45]: Add new modules here after typing

File: xknx/telegram/address.py [hunk 2]
```diff
@@ -81,29 +80,29 @@ class PhysicalAddress(BaseAddress):
         r"^(?P<area>\d{1,2})\.(?P<main>\d{1,2})\.(?P<line>\d{1,3})$"
     )
 
-    def __init__(self, address):
+    def __init__(self, address: Union[object, str, int, tuple]) -> None:
```
Comment (marvin-w) [inline xknx/telegram/address.py:83]: We can't use the same Type as the current class unfortunetly :/

Comment (farmio) [reply xknx/telegram/address.py:83]: this is our excuse to drop 3.6
https://stackoverflow.com/a/49872353/4516732
https://www.python.org/dev/peps/pep-0563/

**Review by farmio — APPROVED** [review]
Nice!

**Pull request merged**